storage/innobase/buf/buf0buf.cc

```diff
@@ -2404,6 +2404,81 @@ buf_zip_decompress(
 	return(FALSE);
 }
 
+ATTRIBUTE_COLD
+/** Try to merge buffered changes to a buffer pool page.
+@param block     buffer-fixed and latched block
+@param rw_latch  RW_X_LATCH, RW_SX_LATCH, RW_S_LATCH held on block
+@param err       error code
+@return whether the page is invalid (corrupted) */
+static bool buf_page_ibuf_merge_try(buf_block_t *block, ulint rw_latch,
+                                    dberr_t *err)
+{
+  ut_ad(block->page.lock.have_any());
+  ut_ad(block->page.buf_fix_count());
+
+  if (fil_page_get_type(block->page.frame) != FIL_PAGE_INDEX ||
+      !page_is_leaf(block->page.frame))
+    return false;
+
+  uint32_t state;
+
+  if (rw_latch != RW_X_LATCH)
+  {
+    if (rw_latch == RW_S_LATCH)
+    {
+      if (!block->page.lock.s_x_upgrade_try())
+      {
+        block->page.lock.s_unlock();
+        block->page.lock.x_lock();
+        state= block->page.state();
```

Comment:
We can declare the variable state here. It is not used outside the scope.

Reply:
Right, we can, but the initialization must be separate from the definition, because otherwise a warning would be emitted for the `goto fail` that leads into this scope.